src/python/pants/backend/python/util_rules/pex_environment.py

```diff
@@ -123,6 +126,7 @@ def environment_dict(self, *, python_configured: bool) -> Mapping[str, str]:
             PATH=create_path_env_var(self.path),
             PEX_INHERIT_PATH="false",
             PEX_IGNORE_RCFILES="true",
+            PEX_ROOT=os.path.join(self.named_caches_dir, "pex_root"),
```

Comment:
This requires staying in sync with https://github.com/pantsbuild/pants/blob/67faec549ef9b57c5b0d211ba7f742f7c9d2a629/src/python/pants/backend/python/util_rules/pex_cli.py#L128

Not sure if it's worth calling that out, or factoring up a general variable. I'm a bit confused why that line is using `.cache`.

Reply:
Yeah, I'm not sure why that is hardcoded.

Reply:
Hrm, so - this is devious.

Given the comments above I'm not sure there was awareness of how this is actually working. This is despite @stuhood pointing out in comments below how this all works on the engine side by symlinking `{process executuon chroot}/{pex_root_path} -> ~/.cache/pants/named_caches/{cache_name}`. 

> This requires staying in sync with 

The `pex_root_path` actually has nothing to do with this. The "pex_root" that's being joined to `self.named_caches_dir` - which is an absolute path! - has to correspond to the cache name key in this line lower down:
https://github.com/pantsbuild/pants/blob/67faec549ef9b57c5b0d211ba7f742f7c9d2a629/src/python/pants/backend/python/util_rules/pex_cli.py#L165

Since `PEX_ROOT` is being exported as an absolute path, no runtime `PexProcess` needs to declare an `append_only_caches` like we do for the Pex CLI at buildtime. This is sneaky. I was certainly confused by it when looking to add a new append only cache to a `PexProcess` and finding it had no such field and no such field was set in the corresponfing `Process` by the rule that converts `PexProcess` to `Process`!

Reply:
And this bit us with remote execution. We can't be using this sneak since the base directory of the named caches dir can shift from execution to execution. We need to plumb the relative path of `.cache/pex_root`.